### PR TITLE
Mvh/nested renamed flavors

### PR DESCRIPTION
**Pull request by mvahowe**
This PR implements #110 (nested JSON for flavorType and flavor, plus renamed flavors). It consists of an updated type schema file plus a modified and renamed test document. It also removes the 0.1 JSON for other flavors in the interests of clarity about how much work still needs to be done.

### CODE REVIEW

**Comment by jag3773** [comment]
@mvahowe did you intend to remove the `$$target`` definitions in schema/type.schema.json ?  Those are there for the automatic sphinx documentation generator.

**Comment by mvahowe** [comment]
@jag3773 No, that wasn't part of my plan - good catch! Can you put them back?

File: schema/type.schema.json [hunk 1]
```diff
@@ -101,154 +82,116 @@
...
-            },
-            "additionalProperties": false,
-            "minProperties": 1,
-            "allOf": [
-                {"not": {"required": ["OT", "OT+"]}},
-                {"not": {"required": ["DC", "OT+"]}}
-            ],
-            "$comment": "OT or DC may not be specified if OT+ is present."
-        }
+	    "properties": {
```
Comment (rdb) [inline schema/type.schema.json]: There's something funny about the indent here; I think a tab character may have gotten in here by accident?

File: schema/type.schema.json [hunk 2]
```diff
@@ -101,154 +82,116 @@
...
+				"name": {
+				    "type": "string",
+				    "enum": [
+					"TextTranslation",
+					"AudioTranslation",
+					"SignLanguageTranslation",
+					"BrailleScripturePublication",
+					"TextStories",
+					"WordAlignment",
+					"Versification"
```
Comment (rdb) [inline schema/type.schema.json]: Let's make this lowerCamelCase.  The notion that this is "kind of like a class" isn't an obvious reason to break consistency to me.

File: docs/examples/artifacts/TextTranslation.json [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+        }
+    },
+    "confidentiality": {
+        "metadata": "unrestricted",
+        "source": "private",
+        "publications": "restricted"
+    },
+    "type": {
+        "flavorType": {
+	    "name": "scripture",
```
Comment (rdb) [inline docs/examples/artifacts/TextTranslation.json]: Funny indent, looks like an accidental tab character.  The indent is also inconsistent further down; maybe you could configure your editor to 4 spaces (or set it to auto-detect).  We could put an `.editorconfig` file in the repo.

**Comment by jag3773** [comment]
@mvahowe and @rdb I fixed the tabs and I re-added the $$target and title information. @mvahowe over to you on the camelCase issue.

**Comment by mvahowe** [comment]
@rdb @jag3773 I had a brief exchange with Dan about this elsewhere. I don't care that much about capitalization, but I don't think we should treat all enums as if they have no existence beyond the schema. We're presumably not going to change country codes to "nl" and "uS". JS uses classcase for classes, and I think encouraging capitalization of flavor names in general will help to avoid "oh, I thought you meant anything that happens to be a text translation" conversations. Nothing we do in the schema provides a magic fix for this, but, in any case, I don't think that "consistent across all enums" is a clinching argument.

(On indents, that's what emacs did. If anyone has a plan for making it use spaces I'll try it.)

**Comment by jag3773** [comment]
For indents, maybe this helps https://stackoverflow.com/questions/69934/set-4-space-indent-in-emacs-in-text-mode @mvahowe?

**Comment by rdb** [comment]
Or this: https://github.com/editorconfig/editorconfig-emacs
(In conjunction with placing an `.editorconfig` file in the root of this repo)

**Comment by mvahowe** [comment]
I changed the case of flavor names, as well as the name of the example doc. (I also grabbed one of the 50 conflicting emacs tips on controlling spaces, so we'll see how that goes.) @jag3773 are we good to merge?

**Comment by mvahowe** [comment]
@rdb You were too fast for me. 49 conflicting Emacs recipes to go. I removed all tabs using another editor and also fixed some indentation of the example document.

**Comment by mvahowe** [comment]
I couldn't find a definition of the idServer prefixes so I left the string in the new idServer field open.

**Comment by jag3773** [comment]
Yes, @mvahowe lets merge this before our meeting today so it's easier for others to contribute with new PRs.

**Pull request merged**